pyramus/src/main/java/fi/otavanopisto/pyramus/webhooks/Webhooks.java

```diff
@@ -1,272 +1,102 @@
-  }
-  
-  public synchronized void onCourseUpdatedBeforeCompletion(@Observes(during=TransactionPhase.BEFORE_COMPLETION) CourseUpdatedEvent event) {
-    datas.addUpdatedCourseId(event.getCourseId());
-  }
+  public void sendWebhookNotification(WebhookPayload<?> payload) {
+    if (!webhooks.isEmpty()) {
+      ObjectMapper objectMapper = new ObjectMapper();
+      try {
+        String data = objectMapper.writeValueAsString(payload);
 
-  public synchronized void onCourseUpdatedAfterSuccess(@Observes(during=TransactionPhase.AFTER_SUCCESS) CourseUpdatedEvent event) {
-    List<Long> updatedCourseIds = datas.retrieveUpdatedCourseIds();
-    
-    for (Long updatedCourseId : updatedCourseIds) {
-      webhookController.sendWebhookNotifications(webhooks, new WebhookCourseUpdatePayload(updatedCourseId));
+        for (Webhook webhook : webhooks) {
+          FutureTask<Boolean> futureTask = new FutureTask<>(new NotificationCallable(webhook.getUrl(), webhook.getSignature(), data));
+          executorService.execute(futureTask);
+        }
+      } catch (JsonProcessingException e) {
+        logger.log(Level.SEVERE, "Failed to send webhook notifications", e);
+      }
     }
   }
-
-  public void onCourseArchived(@Observes(during=TransactionPhase.AFTER_SUCCESS) CourseArchivedEvent event) {
-    webhookController.sendWebhookNotifications(webhooks, new WebhookCourseArchivePayload(event.getCourseId()));
-  }
-  
-  /* Person */
-
-  public void onPersonCreated(@Observes(during=TransactionPhase.AFTER_SUCCESS) PersonCreatedEvent event) {
-    webhookController.sendWebhookNotifications(webhooks, new WebhookPersonCreatePayload(event.getPersonId()));
-  }
-
-  public void onPersonUpdated(@Observes(during=TransactionPhase.AFTER_SUCCESS) PersonUpdatedEvent event) {
-    webhookController.sendWebhookNotifications(webhooks, new WebhookPersonUpdatePayload(event.getPersonId()));
-  }
-
-  public void onPersonArchived(@Observes(during=TransactionPhase.AFTER_SUCCESS) PersonArchivedEvent event) {
-    webhookController.sendWebhookNotifications(webhooks, new WebhookPersonArchivePayload(event.getPersonId()));
-  }
-
-  /* Student */
   
-  public void onStudentCreated(@Observes(during=TransactionPhase.AFTER_SUCCESS) StudentCreatedEvent event) {
-    webhookController.sendWebhookNotifications(webhooks, new WebhookStudentCreatePayload(event.getStudentId()));
-  }
-  
-  public synchronized void onStudentUpdatedBeforeCompletion(@Observes(during=TransactionPhase.BEFORE_COMPLETION) StudentUpdatedEvent event) {
-    datas.addUpdatedStudentId(event.getStudentId());
-  }
+  private ExecutorService executorService;
+  private List<fi.otavanopisto.pyramus.webhooks.Webhook> webhooks;
 
-  public synchronized void onStudentUpdatedAfterSuccess(@Observes(during=TransactionPhase.AFTER_SUCCESS) StudentUpdatedEvent event) {
-    List<Long> updatedStudentIds = datas.retrieveUpdatedStudentIds();
+  private class NotificationCallable implements Callable<Boolean> {
```

Comment:
Could this just be a method, instead of an inner class?

Reply:
I mean like this:

    FutureTask<Boolean> futureTask = new FutureTask(() -> notify(webhook.getUrl(), webhook.getSignature(), data);
    /* ... */
    Boolean notify(String url, String signature, String data) {
        /* ... */
    }